hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileUtil.java

```diff
@@ -398,6 +398,13 @@ public static boolean copy(FileSystem srcFS, FileStatus srcStatus,
                              Configuration conf) throws IOException {
     Path src = srcStatus.getPath();
     dst = checkDest(src.getName(), dstFS, dst, overwrite);
+
+    if (dst != null && src != null &&
+        src.toUri().equals(dst.toUri())) {
```

Comment:
Can we use :
`srcFS.makeQualified(src).equals(dstFS.makeQualified(dst))`
instead of uri comparison, As I said on Jira, this URI comparison isn't working with HDFS